core/templates/dev/head/pages/exploration_player/SupplementalCardDirective.js

```diff
@@ -28,24 +28,34 @@ oppia.directive('supplementalCard', [
         '/pages/exploration_player/' +
         'supplemental_card_directive.html'),
       controller: [
-        '$scope', '$window', 'oppiaPlayerService',
-        'playerPositionService', 'playerTranscriptService',
-        'ExplorationObjectFactory', 'windowDimensionsService',
-        'CONTENT_FOCUS_LABEL_PREFIX', 'TWO_CARD_THRESHOLD_PX',
-        'EVENT_ACTIVE_CARD_CHANGED', 'CONTINUE_BUTTON_FOCUS_LABEL',
+        '$scope', '$timeout', '$window', 'HintManagerService',
+        'oppiaPlayerService', 'playerPositionService',
+        'playerTranscriptService', 'ExplorationObjectFactory',
+        'windowDimensionsService', 'CONTENT_FOCUS_LABEL_PREFIX',
+        'TWO_CARD_THRESHOLD_PX', 'EVENT_ACTIVE_CARD_CHANGED',
+        'CONTINUE_BUTTON_FOCUS_LABEL', 'HINT_REQUEST_STRING_I18N_IDS',
+        'DELAY_FOR_HINT_FEEDBACK_MSEC',
         function(
-          $scope, $window, oppiaPlayerService,
-          playerPositionService, playerTranscriptService,
-          ExplorationObjectFactory, windowDimensionsService,
-          CONTENT_FOCUS_LABEL_PREFIX, TWO_CARD_THRESHOLD_PX,
-          EVENT_ACTIVE_CARD_CHANGED, CONTINUE_BUTTON_FOCUS_LABEL) {
+          $scope, $timeout, $window, HintManagerService,
+          oppiaPlayerService, playerPositionService,
+          playerTranscriptService, ExplorationObjectFactory,
+          windowDimensionsService, CONTENT_FOCUS_LABEL_PREFIX,
+          TWO_CARD_THRESHOLD_PX, EVENT_ACTIVE_CARD_CHANGED,
+          CONTINUE_BUTTON_FOCUS_LABEL, HINT_REQUEST_STRING_I18N_IDS,
+          DELAY_FOR_HINT_FEEDBACK_MSEC) {
           var updateActiveCard = function() {
             var index = playerPositionService.getActiveCardIndex();
             if (index === null) {
               return;
             }
             $scope.activeCard = playerTranscriptService.getCard(index);
             $scope.clearHelpCard();
+            HintManagerService.reset(oppiaPlayerService.getInteraction(
+              $scope.activeCard.stateName).hints);
+
+            $scope.currentInteractionHintsLength = (
```

Comment:
Here and in tutor card: all you use is whether there are hints or not. So, do:

    $scope.hintsExist = Boolean(...);

and just use that instead. No need to store the length.